plugins/bigcommerce/src/data-plugin.ts

```diff
@@ -0,0 +1,160 @@
+//     switch (resource) {
+//         case 'product':
+//             return resourceId ? `products/${resourceId}` : 'products';
+//         case 'category':
+//             return resourceId ? `categories/${resourceId}` : 'categories';
+//         default:
+//             throw new Error('Unsupported resource type');
+//     }
+// };
+
+
+const RESOURCE_TYPES: {
+    name: string;
+    id: BigCommerceResourceType;
+    description: string;
+}[] = [
+        {
+            name: 'Product',
+            id: 'product',
+            description: 'All of your BigCommerce products.',
+        },
+        {
+            name: 'Category',
+            id: 'category',
+            description: 'All of your BigCommerce categories.',
+        },
+    ];
+
+interface DataPluginConfig extends APIOperations {
+    name: string;
+    icon: string;
+}
+
+export const getDataConfig = (service: CommerceAPIOperations): DataPluginConfig => {
+    return {
+        name: 'BigCommerce',
+        icon: 'https://iili.io/JnTyc4s.png',
+        getResourceTypes: async () =>
+            RESOURCE_TYPES.map(
+                (model): ResourceType => ({
+                    ...model,
+                    inputs: () => [
+                        {
+                            friendlyName: 'limit',
+                            name: 'limit',
+                            type: 'number',
+                            defaultValue: 10,
+                            max: 250, // Reflecting BigCommerce's API limit
+                            min: 1,
+                        },
+                        {
+                            friendlyName: 'Search',
+                            name: 'query',
+                            type: 'string',
+                        },
+                    ],
+
+                    toUrl: ({ entry, query, limit }) =>
+                        //this request fails due to lack of auth headers but doesn't seem to affect the functionality
+                        buildBigCommerceUrl({
+                            query,
+                            limit,
+                            resource: model.id,
+                            resourceId: entry,
+                        }),
+                    canPickEntries: true,
+                })
+            ),
+        getEntriesByResourceType: async (resourceTypeId, options = {}) => {
+            // Fetch entries from BigCommerce using the provided service
+            const entry = options.resourceEntryId;
+
+            if (entry) {
+                //get the product or category's data
+                const entryObj = await service[resourceTypeId].findById(entry);
+
+                //filter out undefined properties
+                const newObj = filterOutUndefinedProperties(entryObj)
+
+                //set the product or category data to state
+                appState.designerState.editingContentModel.data.set('state', { [resourceTypeId]: newObj })
+
```

Comment:
```suggestion
```

this is not the way to add data objects to the state, you'd want to fix the HTTP request url above